Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.sb.in

```diff
@@ -22,8 +22,308 @@
+
+    ;; System and user preferences
+    (home-literal "/.CFUserTextEncoding"))
+
+(allow file-read-data
+    (literal "/usr/local/lib/log")) ; <rdar://problem/36629495>
+
+;; Preferences support
+(shared-preferences-read
+    "com.apple.LaunchServices"
+    "com.apple.crypto")
+
+(allow-reading-global-preferences)
+
+; FIXME: This is needed for some security framework calls (that use non-CFPreferences readers)
+(allow file-read-data
+    (literal "/Library/Preferences/com.apple.security.plist")
+    (home-subpath "/Library/Preferences/com.apple.security.plist"))
+
+;; On-disk WebKit2 framework location, to account for debug installations outside of /System/Library/Frameworks,
+;; and to allow issuing extensions.
+(allow-read-directory-and-issue-read-extensions (param "WEBKIT2_FRAMEWORK_DIR"))
+
+#if USE(SYSTEM_CONTENT_PATH)
+(map allow-read-directory-and-issue-read-extensions
+    read-directory-and-issue-read-extension-secondary-paths)
+#endif
+
+(allow mach-lookup
+#if PLATFORM(MAC)
+    (with report) (with telemetry)
+#endif
+#if !ENABLE(CFPREFS_DIRECT_MODE)
+    (global-name "com.apple.cfprefsd.agent")
+#endif
+    (global-name "com.apple.system.logger")
+)
+
+;; Various services required by AppKit and other frameworks
+(allow mach-lookup
+    (global-name "com.apple.SystemConfiguration.configd")
+#if !ENABLE(CFPREFS_DIRECT_MODE)
+    (global-name "com.apple.cfprefsd.daemon")
+#endif
+    (global-name "com.apple.logd")
+    (global-name "com.apple.logd.events")
+)
+
+(deny mach-lookup (with no-log)
+    (global-name "com.apple.CoreServices.coreservicesd")
+    (global-name "com.apple.DiskArbitration.diskarbitrationd")
+    (global-name "com.apple.ViewBridgeAuxiliary")
+    (global-name "com.apple.system.opendirectoryd.membership"))
+
+;; CoreFoundation. We don't import com.apple.corefoundation.sb, because it allows unnecessary access to pasteboard.
+(allow mach-lookup
+    (global-name-prefix "com.apple.distributed_notifications")
+#if !HAVE(CSCHECKFIXDISABLE)
+    (global-name "com.apple.CoreServices.coreservicesd")
+#endif
+)
+
+(deny file-write-create (vnode-type SYMLINK))
+
+;; Reserve a namespace for additional protected extended attributes.
+(deny file-read-xattr file-write-xattr (xattr-prefix "com.apple.security.private."))
+
+;; Analytics.
+(allow mach-lookup (global-name "com.apple.analyticsd"))
+(allow mach-lookup (global-name "com.apple.awdd"))
+
+;; Daemon prefs.
+(allow user-preference-read user-preference-write
+    (preference-domain "com.apple.webkit.webpushd"))
+
+;; Push notification registration.
+(allow mach-lookup
+    (global-name "com.apple.apsd")
+    (global-name "com.apple.windowserver.active")) ;; for -[APSSystemUser isLoggedIn]
+
+;; Allow webpushd to launch UI processes in response to a push via LaunchServices.
+(allow lsopen)
+
+(allow mach-lookup
+    (global-name "com.apple.coreservices.launchservicesd")
+    (global-name "com.apple.coreservices.quarantine-resolver")
+    (global-name "com.apple.lsd.mapdb"))
```

Comment:
I think it would be good to investigate if these are strictly needed, although I think that can be done as a follow-up.

Reply:
You can also add (with telemetry) or (with telemetry-backtrace) for these in order to understand if/when they are in use.

Reply:
These were needed to be able to use the LS API that launches an app.